codec-http2/src/main/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributor.java

```diff
@@ -111,6 +182,65 @@ public void updateStreamableBytes(StreamState state) {
     }
 
     @Override
+    public void updateDependencyTree(int childStreamId, int parentStreamId, short weight, boolean exclusive) {
+        if (weight < MIN_WEIGHT || weight > MAX_WEIGHT) {
+            throw new IllegalArgumentException(String.format(
+                    "Invalid weight: %d. Must be between %d and %d (inclusive).", weight, MIN_WEIGHT, MAX_WEIGHT));
+        }
+        if (childStreamId == parentStreamId) {
+            throw new IllegalArgumentException("A stream cannot depend on itself");
+        }
+
+        State state = state(childStreamId);
+        if (state == null) {
+            if (maxStateOnlySize == 0) {
+                return;
+            }
+            state = new State(childStreamId);
+            stateOnlyRemovalQueue.add(state);
+            stateOnlyMap.put(childStreamId, state);
+        }
+
+        State newParent = state(parentStreamId);
+        if (newParent == null) {
+            if (maxStateOnlySize == 0) {
```

Comment:
same comment as above.  Maybe this code should be made common, since it's basically a copy & paste?